flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniClusterConfiguration.java

```diff
@@ -96,11 +101,20 @@ public void setCommonRpcBindAddress(String bindAddress) {
 		this.commonBindAddress = bindAddress;
 	}
 
+	public void setManagedMemoryPerTaskManager(long managedMemoryPerTaskManager) {
+		checkArgument(managedMemoryPerTaskManager > 0, "must have more than 0 MB of memory for the TaskManager.");
+		this.managedMemoryPerTaskManager = managedMemoryPerTaskManager;
+	}
+
 	// ------------------------------------------------------------------------
 	//  getters
 	// ------------------------------------------------------------------------
 
 	public Configuration getConfiguration() {
+		// update the memory in case that we've changed the number of components (TM, RM, JM)
+		long memory = calculateManagedMemoryPerTaskManager();
```

Comment:
After this method has been called, you can't change the memory configuration anymore because the config value will prevent new calculation in `calculateManagedMemoryPerTaskManager`. Is that desired?